### DISCUSSION TITLE
[How-to] Build RSS Guard for Ubuntu/Debian based systems

### DISCUSSION THREAD

**Discussion by a-pav** [martinrotter/rssguard#1007] [General]
Hi.

Unfortunately, as of now, the most [recent version of RSS Guard available on official Ubuntu repositories is `4.0.4`](https://repology.org/project/rssguard/versions). But  RSS Guard  recently hit `4.4.0` and there's a lot of improvements and newly added features that `4.0.4` lacks.

To my surprise, building RSS Guard wans't that much of a complicated task to do. At least not when I had all necessary dependencies installed!!

So I write the steps I took here for anyone interested.

If you have already installed RSS Guard on your system using Ubuntu's repository, remove the program by the following command. This won't remove your feeds/subscriptions. But have an export of those beforehand just in case.

```
sudo apt purge rssguard
```

And install these dependencies:

```bash
sudo apt install cmake
sudo apt install qtbase5-dev
sudo apt install qtwebengine5-dev
sudo apt install qtmultimedia5-dev
sudo apt install qttools5-dev
sudo apt install qtdeclarative5-dev

sudo apt install checkinstall # optional but recommended 
```
Using `checkinstall` is the same as using `make install` but it creates a `dpkg` entry for easier uninstallation later. See https://en.wikipedia.org/wiki/CheckInstall.

Then follow these steps to build and install:

```bash
# Pull the changes
git pull origin master --tags

# Checkout to the recent version tag
git checkout 4.4.0

# Create a build directory
mkdir build-dir

# A normal build configuration would be like
cmake -B build-dir -S . -DCMAKE_INSTALL_PREFIX=/usr

# Compile with all available CPU cores
cmake --build build-dir -j$(nproc)

# make
make -C build-dir
# and then install, or rather checkinstall
cd build-dir
sudo checkinstall
```

RSS Guard should now be installed and operational.

Please feel free to ask or add anything to this. Thank you.

**Comment by famod** [comment]
Thanks!

FWIW, on Ubuntu 22.04.1 and for rssguard 4.8.6 I also needed:
> sudo apt install libmpv-dev

and I had to use a slightly modified build config:
> cmake -B build-dir -S . -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_WITH_QT6=OFF -DUSE_SYSTEM_SQLITE=OFF

**Comment by Skif-off** [comment]
I don't like `checkinstall` (Flatpak too), so I use a more canonical way: [rssguard_4.8.6+dfsg-1.debian.tar.xz.zip](https://github.com/user-attachments/files/26135595/rssguard_4.8.6%2Bdfsg-1.debian.tar.xz.zip)
I haven't tried 5.x yet, but probably have to do it myself :(

@famod,  why `-DUSE_SYSTEM_SQLITE=OFF`?

**Comment by Skif-off** [comment]
[rssguard_5.0.4+dfsg-1.debian.tar.xz.zip](https://github.com/user-attachments/files/26169165/rssguard_5.0.4%2Bdfsg-1.debian.tar.xz.zip),
works fine in Xubuntu 24.04 (well, some icons don't look very good, but maybe the Breeze icons look good in a dark theme(s) only, I don't know).

**Reply by martinrotter** [reply]
You have to set appropriate icon theme, either Breeze or Breeze Dark or even your own or system icon theme.

**Reply by Skif-off** [reply]
@martinrotter, I know, but system theme does not contain all used icons and I don't like some of the "modern" trends in icon painting (including Breeze/Breeze Dark) :)

Maybe later I'll try to extract the icon names from the source code and think about it...

**Reply by martinrotter** [reply]
ok
